docs/source/en/core/development.md

```diff
@@ -1 +1,339 @@
+
+In the meantime, it supports custom debug parameters.
+
+```bash
+$ egg-bin debug --inpsect=9229
+```
+
+- Debug port of `master` is 9229 or 5858 (Legacy Protocol).
+- Debug port of `agent` is fixed to 5800, it is customizable via `process.env.EGG_AGENT_DEBUG_PORT`.
+- Debug port number of `worker` will increase from `master` port number.
+- During developing period, worker will be hot reloaded once code changed, which will cause debug port to be self incrementing. Please refer to the following IDE configuration for auto-reconnecting.
+
+#### Config Environment
+
+App is also started by `env: local` when executing `debug` command. The configuration being read is an object combined from `config.local.js` and `config.unittest.js` files.
+
+#### Debug With [DevTools] 
+
+The latest DevTools only supports [Inspector Protocol]. Thus you will need to install Node.js 8.x or higher verions to be able to use it.
+
+Execute `npm run debug` to start it:
+
+```bash
+➜  showcase git:(master) ✗ npm run debug
+
+> showcase@1.0.0 debug /Users/tz/Workspaces/eggjs/test/showcase
+> egg-bin debug
+
+Debugger listening on ws://127.0.0.1:9229/f8258ca6-d5ac-467d-bbb1-03f59bcce85b
+For help see https://nodejs.org/en/docs/inspector
+2017-09-14 16:01:35,990 INFO 39940 [master] egg version 1.8.0
+Debugger listening on ws://127.0.0.1:5800/bfe1bf6a-2be5-4568-ac7d-69935e0867fa
+For help see https://nodejs.org/en/docs/inspector
+2017-09-14 16:01:36,432 INFO 39940 [master] agent_worker#1:39941 started (434ms)
+Debugger listening on ws://127.0.0.1:9230/2fcf4208-4571-4968-9da0-0863ab9f98ae
+For help see https://nodejs.org/en/docs/inspector
+9230 opened
+Debug Proxy online, now you could attach to 9999 without worry about reload.
+DevTools → chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:9999/__ws_proxy__
+```
+
+And then choose one of the following ways:
+- Visit the `DevTools` URL printed in the last few lines in console directly. Since this URL is proxied from worker, you don't need to worry about restarting.
+- Open `chrome://inspect`, and config port accordingly, then click `Open dedicated DevTools for Node` to open debug console.
+
+![DevTools](https://user-images.githubusercontent.com/227713/30419047-a54ac592-9967-11e7-8a05-5dbb82088487.png)
+
+#### Debug With WebStorm
+
+`egg-bin` will read environment variable `$NODE_DEBUG_OPTION` set in WebStorm debug mode.
+
+Start npm debug in WebStorm：
+
+![WebStorm](https://user-images.githubusercontent.com/227713/30423086-5dd32ac6-9974-11e7-840f-904e49a97694.png)
+
+#### Debug With [VSCode]
+
+There are 2 ways:
+
+1st method: open settings in VSCode, turn on `Debug: Toggle Auto Attach`, and then execute `npm run debug` in Terminal.
+
+2nd method: setup `.vscode/launch.json` in VSCode, and then simply start with F5 key. (Note that this method will need to turn off the settings metioned in the 1st method).
```

Comment:
(Note: You have to turn off the settings **mentioned** in the 1st method before doing it)